LgiHdmiCec/LgiHdmiCec.cpp

```diff
@@ -0,0 +1,1190 @@
+        }
+
+        void LgiHdmiCec::onDeviceStatusChanged(IARM_EventId_t eventId, const void* data_ptr, size_t len)
+        {
+            const IARM_Bus_CECHost_DeviceStatusChanged_EventData_t* eData = static_cast<const IARM_Bus_CECHost_DeviceStatusChanged_EventData_t*>(data_ptr);
+
+            LOGINFO("id=%d, len=%u", (int)eventId, (unsigned)len);
+
+            if ((eData->logicalAddress == LogicalAddress::TV) || (eData->logicalAddress == LogicalAddress::AUDIO_SYSTEM))
+            { // the scanner may send events for other devices, so skip them ...
+                bool update = false;
+                try
+                {
+                    readAddresses();
+
+                    switch(eData->changedStatus)
+                    {
+                      case IARM_BUS_CECHost_OSD_NAME:
+                      {
+                          LOGINFO("IARM_BUS_CECHost_OSD_NAME for device %d name %s", eData->logicalAddress, eData->data.osdName);
+                          update = setOSDName(eData->data.osdName, eData->logicalAddress);
+                          break;
+                      }
+                      case IARM_BUS_CECHost_VENDOR_ID:
+                      {
+                          LOGINFO("IARM_BUS_CECHost_VENDOR_ID for device %d vendor id 0x%x", eData->logicalAddress, eData->data.vendorId);
+                          update = setVendorId(eData->data.vendorId, eData->logicalAddress);
+                          break;
+                      }
+                      case IARM_BUS_CECHost_POWER_STATUS:
+                      {
+                          LOGINFO("IARM_BUS_CECHost_POWER_STATUS for device %d powerState %d", eData->logicalAddress, eData->data.powerState);
+                          update = setPowerState(eData->data.powerState, eData->logicalAddress);
+                          break;
+                      }
+                      case IARM_BUS_CECHost_CONNECT_STATUS:
+                      {
+                          LOGINFO("IARM_BUS_CECHost_CONNECT_STATUS for device %d isConnected %d", eData->logicalAddress, eData->data.isConnected);
+                          update = setConnectedState(eData->data.isConnected, eData->logicalAddress);
+                          break;
+                      }
+		      case IARM_BUS_CECHost_AUDIO_MODE:
+		      {
+			  LOGINFO("IARM_BUS_CECHost_AUDIO_MODE systemAudioMode %d\n", eData->data.systemAudioMode);
+			  update = setSystemAudioMode(eData->data.systemAudioMode);
+			  break;
+		      }
+                      default:
+                      {
+                          LOGWARN("Unsupported event IARM_BUS_CECHost %d !!!!!", eData->changedStatus);
+                          break;
+                      }
+                    }
+                }
+                catch (const std::exception& e)
+                {
+                    LOGERR("exception caught");
+                }
+                if (update)
+                {
+                    m_updated = true;
```

Comment:
m_update is not guaranteed to be changed atomically. Shouldn't we use some mutex to protect this?